build.psake.ps1

```diff
@@ -134,6 +134,41 @@ Task Sign -depends BuildImpl -requiredVariables SettingsPath, SignScripts {
     }
 }
 
+Task Analyze -depends Build -requiredVariables CodeAnalysisStop, OutDir {
```

Comment:
I wonder if we should change the Test task to depend on the Analyze task instead of the Build task?  IOW script analysis needs to pass first (assuming action set to error) before we are at a point where the Test task should run?  And since, Publish depends on a successful Test that would mean that Publish would also require a successful Analyze task.  

Another thought is that Analyze should be part of the Build task flow - perhaps between BuildImpl and Sign?  You might want that if you set the Action to Error ie it is more a build failure so no point in signing files.